rewrite-test/src/main/kotlin/org/openrewrite/java/tree/TreeBuilderTest.kt

```diff
@@ -48,6 +47,116 @@ interface TreeBuilderTest {
         assertTrue(snippets[0] is J.MethodInvocation)
     }
 
+    @Test
+    fun buildSnippetLocalVariableReference(jp: JavaParser) {
+        val a = jp.parse("""
+            import java.util.List;
+            public class A {
+                int n = 0;
+                
+                void foo(String m, List<String> others) {
+                    n++;
+                }
+            }
+        """.trimIndent())[0]
+
+        val method = a.classes[0].methods[0]
+        val methodBodyCursor = RetrieveCursor(method.body).visit(a)
+        val paramName = (method.params.params[0] as J.VariableDecls).vars[0].name
+
+        val snippets = TreeBuilder(a).buildSnippet<Statement>(
+                methodBodyCursor,
+                """
+                    n++;
+                    others.add(${paramName.printTrimmed()});
+                 """)
+        val unary : Expression = snippets[0] as Expression
+        assertEquals(unary.type, JavaType.Primitive.Int,"The unary should have a type of int.")
+
+        //Second Statement should be "others.add(m)" and this should be a Unary that has type information.
+        assertTrue(snippets[1] is J.MethodInvocation)
```

Comment:
The comment says that the second statement should have type information, but the test doesn't actually assert that it does. That assertion should be added.